server/controllers/finance/reports/unpaid-invoice-payments/index.js

```diff
@@ -51,7 +51,11 @@ async function getUnbalancedInvoices(options) {
     new Date(options.dateTo),
   ];
 
-  const wherePart = options.debtorGroupName ? `WHERE debtorGroupName = ${db.escape(options.debtorGroupName)}` : '';
+  const { debtorGroupName, serviceId } = options;
+  let wherePart = debtorGroupName ? `WHERE debtorGroupName = ${db.escape(debtorGroupName)}` : '';
+  if (serviceId) {
+    wherePart = (wherePart.length < 2) ? `WHERE serviceID=${serviceId}` : `${wherePart} AND serviceId=${serviceId}`;
+  }
```

Comment:
👍 very good

Is it possible to have on the view (handlebars) the selected service if it exists